### PR TITLE
fix(tiller): Avoid corrupting storage via a lock

### PR DESCRIPTION
**Pull request by cristiklein**
If two `helm upgrade`s are executed at the exact same time, then one of
the invocations will fail with "already exists".

If one `helm upgrade` is executed and a second one is started while the
first is in `pending-upgrade`, then the second invocation will create a
new release. Effectively, two helm invocations will simultaneously
change the state of Kubernetes resources -- which is scary -- then two
releases will be in `deployed` state -- which can cause other issues.

This commit fixes the corrupted storage problem, by introducting a poor
person's lock. If the last release is in a pending state, then helm will
abort. If the last release is in a pending state, due to a previously
killed helm, then the user is expected to do `helm rollback`.

This is a port to Helm v2 of #7322.

Signed-off-by: Cristian Klein <cristian.klein@elastisys.com>

### CODE REVIEW

**Review by bacongobbler — APPROVED** [review]
LGTM

**Comment by bacongobbler** [comment]
Going to hold off on merging until we have another maintainer available to review. I think this should be fine but I'd love to have another maintainer check and confirm.

**Comment by paulczar** [comment]
:+1: seems like a nice lightweight way to handle the bug, especially with it approved and ready to merge in v3 I don't see any reason to hold this off.

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @cristiklein

**Pull request merged**
